src/main/java/org/springframework/samples/petclinic/rest/controller/PetRestController.java

```diff
@@ -95,5 +96,11 @@ public ResponseEntity<PetDto> deletePet(Integer petId) {
         return new ResponseEntity<>(HttpStatus.NO_CONTENT);
     }
 
-
+    @PreAuthorize("hasRole(@roles.OWNER_ADMIN)")
+    @Transactional
```

Comment:
Transaction management is handle at the service layer.
The `addPet` is already transactional.

Reply:
OK, it makes sense.

I also removed the transaction management annotation from the deletePet method

Reply:
Very strange, I don't see the comment I received by mail this mornin at 10:15
```
OK, it makes sense.
I also removed the transaction management annotation from the deletePet method
```

Reply:
I tried to fix it using codespaces and I had an issue w/ this endpoint.

I will try to do that on my laptop

Reply:
It's fixed now for both deletePet and addPet